src/main/scala/com/danielasfregola/twitter4s/http/clients/streaming/statuses/TwitterStatusClient.scala

```diff
@@ -18,62 +20,72 @@ trait TwitterStatusClient {
-    *                   https://dev.twitter.com/streaming/overview/request-parameters#follow</a>
-    * @param tracks : Empty by default. Keywords to track. Phrases of keywords are specified by a comma-separated list.
-    *                For more information <a href="https://dev.twitter.com/streaming/overview/request-parameters#track" target="_blank">
-    *                  https://dev.twitter.com/streaming/overview/request-parameters#track</a>
-    * @param locations : Empty by default. Specifies a set of bounding boxes to track.
-    *                    For more information <a href="https://dev.twitter.com/streaming/overview/request-parameters#locations" target="_blank">
-    *                      https://dev.twitter.com/streaming/overview/request-parameters#locations</a>
-    * @param languages : Empty by default. A comma separated list of 'BCP 47' language identifiers.
-    *                    For more information <a href="https://dev.twitter.com/streaming/overview/request-parameters#language" target="_blank">
-    *                      https://dev.twitter.com/streaming/overview/request-parameters#language</a>
+    * @param follow : Empty by default. List of user IDs, indicating the users whose Tweets should be delivered on the stream.
+    * @param tracks : Empty by default. List of phrases which will be used to determine what Tweets will be delivered on the stream.
+    *                 Each phrase must be between 1 and 60 bytes, inclusive.
+    * @param locations : Empty by default. List of longitude,latitude pairs specifying a set of bounding boxes to filter Tweets by.
+    *                    Only geolocated Tweets falling within the requested bounding boxes will be included.
+    *                    Each bounding box should be specified as a pair of longitude and latitude pairs, with the southwest corner of the bounding box coming first.
+    * @param languages : Empty by default. List of 'BCP 47' language identifiers.
     * @param stall_warnings : Default to false. Specifies whether stall warnings (`WarningMessage`) should be delivered as part of the updates.
-    * @param f: the function that defines how to process the received messages
+    * @param filter_level : Default value is none, which includes all available Tweets.
+    *                       Set the minimum value of the filter_level Tweet attribute required to be included in the stream.
+    * @param f : Function that defines how to process the received messages.
+    *
+    * @see <a href="https://developer.twitter.com/en/docs/tweets/filter-realtime/guides/basic-stream-parameters">Basic stream parameters</a>
     */
   def filterStatuses(follow: Seq[Long] = Seq.empty,
                      tracks: Seq[String] = Seq.empty,
-                     locations: Seq[Double] = Seq.empty,
+                     locations: Seq[(Double, Double)] = Seq.empty,
```

Comment:
Probably we should use a case class here rather than simple tuples....but how should be call it? we already have a case class Location (that is quite different) and used in the REST Client

Reply:
There are some improvements that can be made here. Looking at the official twitter java client which is quite outdated [here](https://github.com/twitter/hbc/blob/master/hbc-core/src/main/java/com/twitter/hbc/core/endpoint/Location.java), seems that each bounding box location must be in the form of (southwest coordinate, northwest coordinate).

We can create case classes similar to the link above. Not sure about the name though. For reference, elasticsearch has similar [usecase](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-geo-bounding-box-query.html).

Also, looking at this [doc](https://developer.twitter.com/en/docs/tutorials/filtering-tweets-by-location), most coordinates are in the form of [LONG,LAT], except geo attribute [LAT, LONG], we can improve case classes such as [Area](https://github.com/DanielaSfregola/twitter4s/blob/master/src/main/scala/com/danielasfregola/twitter4s/entities/Area.scala). It would be another breaking change.

Reply:
I wouldn't worry too much about breaking changes: as long as we document them and that they are done with reason, then people are gonna be fine with it. The problem is that we need to do them, bump up the version and properly document them.

We have a few requests already that we know are probably gonna need API breaking changes, for example see #142 and #144.

I am gonna move this discussion into a separate issue (see #169) so that we can discuss it more easily.